cartographer_toyota_hsr/configuration_files/toyota_hsr_3d.lua

```diff
@@ -0,0 +1,36 @@
+-- Copyright 2016 The Cartographer Authors
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+include "map_builder.lua"
+
+options = {
+  map_builder = MAP_BUILDER,
+  map_frame = "map",
+  tracking_frame = "base_imu_frame",
+  published_frame = "odom",
+  odom_frame = "odom",
+  provide_odom_frame = false,
+  use_odometry = true,
+  use_laser_scan = false,
+  use_multi_echo_laser_scan = false,
+  num_point_clouds = 1,
+  lookup_transform_timeout_sec = 0.2,
+  submap_publish_period_sec = 0.3,
+  pose_publish_period_sec = 5e-3,
+}
+
+MAP_BUILDER.use_trajectory_builder_3d = true
+MAP_BUILDER.sparse_pose_graph.optimization_problem.huber_scale = 5e3
```

Comment:
Did you try the backpack_3d.lua settings? They seem to work well.

I'm not suggesting you load them instead, that won't work obviously. But you may want to copy the MAP_BUILDER configuration from there.